api/src/pcapi/scripts/venue/venue_ban_id/fix_incomplete_ban_id.py

```diff
@@ -0,0 +1,119 @@
+                venue = permanent_venues_ids.get(venue_id)
+                if venue and ban_id:
+                    venue.banId = ban_id
+            except KeyError:
+                print(f"ERROR: Venue ID is missing for this entry: {data}")
+    # Log infos of which venues have been updated
+    venues_with_ban_id = [v for v in permanent_venues if v.banId]
+    venues_without_ban_id = [v for v in permanent_venues if not v.banId]
+    print(
+        f"{len(venues_with_ban_id)} Venues have been updated \n"
+        f"{len(venues_without_ban_id)} Venues not updated. Check in DB with this command:\n"
+        f'> SELECT id FROM venue WHERE "isPermanent" IS TRUE AND "isVirtual" IS FALSE AND "banId" IS NULL\n'
+        f"\n Don't forget to remove file from system when you are done:\n"
+        f"> rm {filename}"
+    )
+
+
+def get_incomplete_ban_ids(venue_list: list[Venue]) -> list[Venue]:
+    venues_with_incomplete_ban_ids = []
+    for venue in venue_list:
+        if not venue.banId:
+            venues_with_incomplete_ban_ids.append(venue)
+        elif venue.banId and venue.address and (len(venue.banId) <= 10 and not venue.address.isalpha()):
+            # For venues which have a short BAN id (meaning BAN id doesn't inform on street number)
+            # We check if address has in fact no street number. If it has, it means the BAN id is incomplete
+            venues_with_incomplete_ban_ids.append(venue)
+    return venues_with_incomplete_ban_ids
+
+
+def execute(dry_run: bool = True) -> None:
+    print("Adding missing or incomplete BAN id to permanent venues")
+
+    # CSV file where we save the ban ids fetched from Adresse API
+    filename = PATH / "ban_ids_from_adresse_api.csv"
+
+    # Query all permanent Venues from DB (That are not virtual)
+    all_permanent_venues = Venue.query.filter(Venue.isPermanent == True, Venue.isVirtual == False).all()
+
+    # In case of dry run, we request the API and save data to the CSV file
+    if dry_run:
+        print(f"Dry run: only fetching incomplete banIds from adresse API and save them in {filename}")
+
+        permanent_venues_with_incomplete_ban_id = get_incomplete_ban_ids(all_permanent_venues)
+        request_adresse_api_adresse(filename, permanent_venues_with_incomplete_ban_id)
+
+    # Once we have saved all data to csv, we write them to DB
+    if not dry_run:
+        if not pathlib.Path(filename).is_file():
+            print("WARNING: Run this script in dry run first")
+            return
+        update_venues_from_csv(filename, all_permanent_venues)
+
+        batches = [all_permanent_venues[i : i + 1000] for i in range(0, len(all_permanent_venues), 1000)]
+        for i, batch in enumerate(batches, start=1):
+            for venue in batch:
+                try:
+                    db.session.add(venue)
+                except sa_exc.DataError as e:
+                    print(f"Error: can't update venue {venue.id}: {e}")
+            db.session.commit()
```

Comment:
Si tu as une erreur lors du commit d'un lot de lieux, la fonction s'arrêtera net. Tu pourrais ajouter quelque chose comme ça pour logger le lot problématique et continuer l'exécution : 

```suggestion
            try:
                db.session.commit()
            except Exception:
                logger.exception("Could not update batch %d", i)
                db.session.rollback()
```